Controllers/RelationController.cs

```diff
@@ -17,73 +13,58 @@ namespace AndersenCoreApp.Controllers
     [ApiController]
     public class RelationController : ControllerBase
     {
-
         private IRelationService relationService;
 
         public RelationController(IRelationService service)
         {
             relationService = service;
         }
 
-        
         [HttpGet]
-        public List<RelationDTO> Get(string orderProperty ="", string orderCategory = "")
+        public ActionResult<List<RelationViewModel>> GetRelationsList([FromQuery] string orderProperty = "", [FromQuery] string categoryName = "")
         {
             var relations = relationService.GetAll();
-            if (!string.IsNullOrEmpty(orderProperty))
+            if (relations == null)
             {
-                relations = relationService.GetSortedListByProperties(relations, orderProperty);
+                return NotFound();
             }
```

Comment:
GetAll() doesn't have to return null. It should return an empty array if there are no entities in db.

Reply:
Do not use GetAll() and than GetSortedAndFilteredRelations(...) 
Use one method **GetRelations(string filterByCategoryName, string sortByProperty, OrderBy orderBy)**